pkg/agent/cniserver/server.go

```diff
@@ -527,6 +529,20 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	}
 	cniVersion := cniConfig.CNIVersion
 	cniResult, _ := result.Result.GetAsVersion(cniVersion)
+	status, err := netdefutils.CreateNetworkStatus(cniResult, cniConfig.Name, true, nil)
+	if err != nil {
+		klog.ErrorS(err, "Create NetworkStatus failed", "Pod", klog.KRef(podNamespace, podName))
+	} else {
+		pod, err := s.kubeClient.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
+		if err == nil {
+			netStatus := []netdefv1.NetworkStatus{*status}
+			if err := netdefutils.SetNetworkStatus(s.kubeClient, pod, netStatus); err != nil {
+				klog.ErrorS(err, "Pod network status annotation update failed", "Pod", klog.KObj(pod))
+			} else {
+				klog.V(2).InfoS("Pod network status annotation updated", "Pod", klog.KObj(pod), "NetworkStatus", netStatus)
+			}
+		}
```

Comment:
What if there is an error when getting the Pod? It seems there is such logic to handle error case.
Should we have some retry logic when error happens? @tnqn @antoninbas 
We probably can ignore not found error and retry for other errors if there is a need.

Reply:
Updated.